src/pages/Environments.js

```diff
@@ -445,7 +451,7 @@ const Environments = () => {
   const getWorkspaceCell = (namespace, name, appType, shouldWarn) => {
     return !!name ?
       h(Fragment, [
-        h(Link, { href: Nav.getLink('workspace-dashboard', { namespace, name }), style: { wordBreak: 'break-word' } }, [name]),
+        h(Link, { href: getLink('workspace-dashboard', { namespace, name }), style: { wordBreak: 'break-word' } }, [name]),
```

Comment:
an alternative to the nav optional prop with .getLink method pattern used here is to flip (remove) the dependency and instead have a required prop (let me specify in TS to give the idea):
export interface WorkspaceClickInfo {
  namespace: string
  name: string
}
export interface EnvironmentsPageProps {
  onWorkspaceClick: (wsInfo: WorkspaceClickInfo) => void
}
A wrapper component here in terra-ui can provide onWorkspaceClick that uses Nav.gotoPath(...) to achieve same effect as getLink does here.  And AoU can just handle the same way but with it's own nav mechanic.

Reply:
Nick helped me interpret what you are asking here. I think I've captured the spirit.

Reply:
some good discussion on this in the mobbing session.  I am ok with this improvement being deferred.